distributed/core.py

```diff
@@ -247,7 +236,9 @@ def set_thread_ident():
             self.thread_id = threading.get_ident()
 
         self.io_loop.add_callback(set_thread_ident)
-        self._startup_lock = asyncio.Lock()
+        self._started = asyncio.Event()
+        self.__status = Status.init
```

Comment:
Reusing the `status` attribute for workers, nannies, schedulers and the core server is not easily possible without some major changes. There are many different semantics encoded and I don't believe the server base class should react to any status change of the child. I don't mind renaming this attribute but I believe there should be different stati